go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -877,3 +884,85 @@ func (bsnc *BaseSecondaryNetworkController) requireDHCP(pod *corev1.Pod) bool {
+				sourcePodLsp := &nbdb.LogicalSwitchPort{Name: sourcePodLspName}
+				sourcePodLsp, err = libovsdbops.GetLogicalSwitchPort(bsnc.nbClient, sourcePodLsp)
+				if err != nil {
+					return nil, fmt.Errorf("cannot fetch switch port for node %s: %v", sourcePodLspName, err)
+				}
+
+				sourcePodLsp.Enabled = ptr.To(false)
+				lsps = append(lsps, sourcePodLsp)
+			}
+
+			logicalSwitch, err := libovsdbops.GetLogicalSwitch(bsnc.nbClient, &nbdb.LogicalSwitch{Name: switchName})
+			if err != nil {
+				return nil, fmt.Errorf("cannot fetch switch for node %s: %v", switchName, err)
+			}
+			ops, err = libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitchOps(bsnc.nbClient, nil, logicalSwitch, lsps...)
+			if err != nil {
+				return nil, fmt.Errorf("error updating logical switch port %+v on switch %+v: %+v", *lsp, *logicalSwitch, err)
+			}
+		}
+	}
+	return ops, nil
+}
+
+func (bsnc *BaseSecondaryNetworkController) updateLogicalSwitchPortForFailedLiveMigration(pod *corev1.Pod, nadName string) error {
+	kubevirtLiveMigrationStatus, err := kubevirt.DiscoverLiveMigrationStatus(bsnc.watchFactory, pod)
+	if err != nil {
+		return fmt.Errorf("failed to discover Live-migration status after pod termination: %w", err)
+	}
+	if kubevirtLiveMigrationStatus != nil {
+		if pod.Name == kubevirtLiveMigrationStatus.TargetPod.Name &&
+			kubevirtLiveMigrationStatus.State == kubevirt.LiveMigrationFailed &&
+			podLspExistsOnSwitch(bsnc.TopologyType(), bsnc.isPodScheduledinLocalZone(kubevirtLiveMigrationStatus.SourcePod)) {
+			// make sure sourcePod lsp is enabled if migration failed after DomainReady was set.
+			sourcePodLspName := bsnc.GetLogicalPortName(kubevirtLiveMigrationStatus.SourcePod, nadName)
+			sourcePodLsp := &nbdb.LogicalSwitchPort{Name: sourcePodLspName}
+			sourcePodLsp, err = libovsdbops.GetLogicalSwitchPort(bsnc.nbClient, sourcePodLsp)
+			if err != nil {
+				return fmt.Errorf("cannot fetch switch port for node %s: %v", sourcePodLspName, err)
+			}
+			sourcePodLsp.Enabled = ptr.To(true)
+
+			logicalSwitchName, err := bsnc.getExpectedSwitchName(pod)
+			if err != nil {
+				return fmt.Errorf("cannot fetch switch name for pod %s: %v", pod.Name, err)
+			}
+			logicalSwitch, err := libovsdbops.GetLogicalSwitch(bsnc.nbClient, &nbdb.LogicalSwitch{Name: bsnc.GetNetworkScopedSwitchName(logicalSwitchName)})
+			if err != nil {
+				return fmt.Errorf("cannot fetch switch for node %s: %v", logicalSwitchName, err)
+			}
+			ops, err := libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitchOps(bsnc.nbClient, nil, logicalSwitch, sourcePodLsp)
+			if err != nil {
+				return fmt.Errorf("error updating logical switch port %+v on switch %+v: %+v", *sourcePodLsp, *logicalSwitch, err)
+			}
+
+			_, err = libovsdbops.TransactAndCheckAndSetUUIDs(bsnc.nbClient, sourcePodLsp, ops)
+			if err != nil {
+				return fmt.Errorf("error transacting operations %+v: %v", ops, err)
+			}
```

Comment:
We should not do this here.

Reply:
this function is supposed to be on the next commit. will fix

Reply:
DONE.
Notice that in the "DeleteResource" path there is no Transact function so I had to add one in this function...